resourcename/revision.go

```diff
@@ -0,0 +1,18 @@
+package resourcename
+
+import "strings"
+
+// HasRevision checks if a resource name has a revision.
+func HasRevision(name string) bool {
+	// TODO: Use Scanner to only return true for valid resource names.
+	indexOfAt := strings.LastIndexByte(name, '@')
+	return indexOfAt > 0 && indexOfAt < len(name)-1
+}
+
+// TrimRevision takes a resource name and returns it without any revision.
+func TrimRevision(name string) string {
+	if indexOfAt := strings.LastIndexByte(name, '@'); indexOfAt != -1 {
+		return name[:indexOfAt]
+	}
+	return name
+}
```

Comment:
This does not handle resource names where the parent is revisioned, for example:
`datasets/1@beef/tables/1 -> datasets/1/tables/1`

Perhaps it should be called `TrimRevisions` to imply that it will trim *all* revisions from the name.

Reply:
Very good point. Let's remove this one for now.